File: iphone/Classes/TextParsedResult.m
//
//  TextParsedResult.m
//  ZXing
//
//  Created by Christian Brunschen on 23/05/2008.
/*
 * Copyright 2008 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#import "TextParsedResult.h"


@implementation TextParsedResult

@synthesize text;

- initWithString:(NSString *)s {
  if ((self  = [super init]) != nil) {
    self.text = s;
  }
  return self;
}

+ parsedResultForString:(NSString *)s {
  return [[[self alloc] initWithString:s] autorelease];
}

+ (NSString *)typeName {
  return @"TEXT";
}

- (NSString *)stringForDisplay {
  return self.text;
}

- (void) populateActions {
}

- (void)dealloc {
  [text release];
  [super dealloc];
}

@end

File: csharp/client/result/TextParsedResult.cs
/*
 * Copyright 2007 ZXing authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

namespace com.google.zxing.client.result
{

	/// <summary>
	/// A simple result type encapsulating a string that has no further
	/// interpretation.
	/// 
	/// @author Sean Owen
	/// </summary>
	public sealed class TextParsedResult : ParsedResult
	{

	  private readonly string text;
	  private readonly string language;

	  public TextParsedResult(string text, string language) : base(ParsedResultType.TEXT)
	  {
		this.text = text;
		this.language = language;
	  }

	  public string Text
	  {
		  get
		  {
			return text;
		  }
	  }

	  public string Language
	  {
		  get
		  {
			return language;
		  }
	  }

	  public override string DisplayResult
	  {
		  get
		  {
			return text;
		  }
	  }

	}

}